components/programmes/ConditionsOfJoining.tsx

```diff
@@ -42,13 +42,13 @@ export function ConditionsOfJoining({
       </SummaryList.Row>
       <SummaryList.Row>
         <SummaryList.Actions style={{ borderBottom: 0 }}>
-          <Button
-            secondary
-            onClick={() => viewCoj(programmeMembershipId, programmeName)}
+          <Link
+            to={`/programmes/${programmeMembershipId}/sign-coj`}
+            onClick={() => setCojState(programmeMembershipId, programmeName)}

```

Comment:
Splitting the state update and navigation appears to work fine, am I correct in thinking the `onClick` will always complete before the navigation?